src/Navigation/Xamarin.Forms/Navigation.Xamarin.Forms/Service/NavigationService.show.cs

```diff
@@ -203,35 +214,89 @@ protected async Task<TViewModel> InitializeViewModelAsync<TViewModel>()
 
 
 
+        private Func<Page, Task> BuildShowAsyncFunc(
+            Page page)
+        {
+            if (RootPage is MasterDetailPage &&
+                page is DetailPage)
+            {
+                return SetDetailAsync;
+            }
+            else if (RootPage is TabbedPage &&
+                page is TabPage)
+            {
+                return SetTabAsync;
+            }
+            else return PushAsync;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'else' block.](https://app.codacy.com/gh/UltimateCodeMonkeys/CodeMonkeysMVVM/pullRequest?prid=5598919)